### ISSUE TITLE
False positives on similar_names

### ISSUE THREAD

**Issue by JohanLorenzo** [Manishearth/rust-clippy#818]
Thank you for this amazing tool!

Today I came across a couple of examples that don't seem legit, at first:
``` rust
warning: binding's name is too similar to existing binding, #[warn(similar_names)] on by default
let (tx1, rx1) = channel();
          ^~~
note: existing binding defined here
let (tx1, rx1) = channel();
     ^~~
```
`(tx, rx)` themselves are not reported.
I encountered another example with variables called `setter` and `getter`. 

I believe we could make clippy less strict. What do you all think?

**Comment by oli-obk** [comment]
I'll add them to the whitelist
